src/extension.ts

```diff
@@ -1,176 +1,208 @@
-  }
-}
-
-function getThemeName(theme: ThemeObject): string {
-  return theme.id || theme.label;
-}
 
 function getUserSettings(): vscode.WorkspaceConfiguration {
   return vscode.workspace.getConfiguration();
 }
 
 function getExtensionConfig(): vscode.WorkspaceConfiguration {
-  return vscode.workspace.getConfiguration('randomThemeSwitcher');
+  return vscode.workspace.getConfiguration(EXTENSION_NAME);
 }
 
-async function changeTheme({
-  extensionConfig = getExtensionConfig(),
-  userSettings = getUserSettings(),
-  context
-}: {
-  context?: vscode.ExtensionContext;
-  extensionConfig?: vscode.WorkspaceConfiguration;
-  userSettings?: vscode.WorkspaceConfiguration;
-} = {}) {
-  const themeList = getThemeList(extensionConfig, userSettings);
+
+async function changeTheme(
+  context?: vscode.ExtensionContext) {
+  isSwitching = true;
+
+  const userSettings = getUserSettings();
   const i = getRandomInt(themeList.length - 1);
   const newTheme = themeList[i];
 
   if (MATERIAL_LIST.findIndex(mat => mat === newTheme) && context !== undefined) {
     await context.globalState.update(LAST_THEME_MATERIAL, true);
   }
 
-  await userSettings.update('workbench.colorTheme', newTheme, true);
-  await vscode.window.showInformationMessage(`Theme switched to ${newTheme}`);
+  userSettings.update('workbench.colorTheme', newTheme, true).then(() => {
+    if (switchMode !== 'typing') {
+      vscode.window.showInformationMessage(`Theme switched to ${newTheme}`);
+    }
+    else {
+      isSwitching = false;
+    }
+  });
+
 }
 
-function getInstalledThemes(): string[] {
-  return vscode.extensions.all.reduce(
-    (acc, ext) => {
-      const themeInfo: Option<Array<ThemeObject>> = fromNullable(ext.packageJSON.contributes.themes);
+async function getInstalledThemes(): Promise<string[]> {
+
+  const promise = new Promise<string[]>((r, c) => {
+    vscode.window.showQuickPick([ThemeTypes.Both, ThemeTypes.Ligth, ThemeTypes.Dark], { placeHolder: "Choose your side, May the code be with you !" }).then(async (choosedSide) => {
+      const excludeRegexPattern = await vscode.window.showInputBox({ prompt: "Exclude themes that match this RegEx, or just Press `Enter` to ignore and continue" });
+      const excludeRegex = excludeRegexPattern ? new RegExp(excludeRegexPattern) : null;
+      if (choosedSide) {
+        let result = vscode.extensions.all.reduce(
+          (acc, ext) => {
+            if (ext && ext.packageJSON && ext.packageJSON.contributes) {
+              const themeInfo: Option<Array<ThemeObject>> = fromNullable(ext.packageJSON.contributes.themes);
+              if (themeInfo._tag !== 'None') {
+                return themeInfo.fold(acc, themes => {
+                  return acc.concat(themes.filter((theme) => choosedSide === ThemeTypes.Both || theme.uiTheme === choosedSide).map((theme) => theme.id || theme.label));
+                });
+              }
+            }
+            // else {
```

Comment:
se non è più usato rimuovi pure

Reply:
(Ciao, non ho mai usato questa funzionalità di GitHub e trovo difficoltoso orientarsi con questi mini riquadri, ho risolto guardando con la mia copia locale.)

L'avevo tenuto solo perchè prima (in Debug) si piantava qui la ricerca dei temi ,per via del  `ext.packageJSON.contributes = null`.
Tolto 👍🏻

Se cliccassi su "Resolve conversation" me lo risolve nel senso che tiene il codice così com'è, lo modifica o chiude solo questa mini chat ?

Nel dubbio, farò una commit finale.

Reply:
chiude solo questa "chat" e non tocca il codice in nessun modo